src/components/Searcher.js

```diff
@@ -1008,13 +1046,38 @@ class Searcher extends React.Component<SearcherDefaultProps, SearcherProps, Sear
     signal.relevancyModelNames = querySignal.relevancyModelNames;
     signal.relevancyModelVersion = querySignal.relevancyModelVersion;
     signal.signalTimestamp = Date.now();
-    signal.ttl = false;
     signal.type = 'facet';
     signal.weight = weight;
 
     new Signals(this.props.baseUri).addRawSignal(signal);
   }
 
+  /**
+   * Add signal each time a promotion is clicked.
+   */
+  addPromotionSignal = (signalDocID: string, signalDocOrdinal: number) => {
+    const querySignal = this.getDefaultQuerySignal(true);
+    if (!querySignal) {
+      return;
+    }
+    const signal = new SignalData();
```

Comment:
This seems like it would be error prone. If the SignalData object is ever changed to add additional fields, they won't be copied here. I would make a `clone()` method in the SignalData class and call it instead, and then just set the values you need to override. (Anyone editing the SignalData class will still need to update the `clone()` method but that will be expected, without having to search for all the uses of SignalData in the code.)